src/shapes/Object/Object.ts

```diff
@@ -1360,7 +1360,15 @@ export class FabricObject<
    * @param {Boolean} [options.viewportTransform] Account for canvas viewport transform
    * @return {HTMLCanvasElement} Returns DOM element <canvas> with the FabricObject
    */
-  toCanvasElement(options: any = {}) {
+  toCanvasElement(
+    options: any = {},
+    createToCanvas = (el: HTMLCanvasElement) =>
```

Comment:
any reason for this function to not be another option?
Also as pointed out the name of the function could be improved, some suggestions
- canvasCreator
- canvasProvider

Reply:
Good points, done